NSTTutorials/RuntimeArguments/README.md

```diff
@@ -0,0 +1,61 @@
+- [TestNG SureFire plugin](https://maven.apache.org/surefire/maven-surefire-plugin/examples/testng.html)
+
+## Runtime argument usage
+
+NST TestNG runtime arguments are specified for NST either as command line arguments, specified in the TestNG Surefire plugin configuration, or optionally as parameters in a TestNG test suite file.
+Various utilitarian runtime arguments are supported in NST when using the `RuntimeConfigManager`. It's up to you how you want to accomplish providing these NST TestNG runtime argument values. One possible disadvantage of hardcoding runtime arguments in the Surefire TestNG configuration is that the pom.xml file may need to be reloaded / refreshed whenever any change is made (depending on your IDE), as opposed to if the arguments were defined in a TestNG Test Suite file, or in the command line, which would take effect immediately.
+
+There are built in getter methods for the "out of the box" runtime arguments that were [linked](../../NST/README.md#Runtime Arguments) above.
+Additionally, there is an override method for the platform that has been set (`overridePlatform`).
+
+In general, the runtime arguments that are supported (without any customization) are sufficient for creating tests with NST. However, there may be use cases wherein you want to add your own custom runtime argument, which is explained below. 
+
+## Adding custom runtime arguments
+
+To add a custom runtime argument that can be read during a test (for example), you can leverage the `RuntimeConfigManager` . 
+
+To create the runtime argument object, create a class that implements `RuntimeConfigValue`. All the runtime arguments above implement this interface, which you can reference as an example.
+
+[Here](src/test/java/com/ebay/nst/tutorials/rest/runtimearguments/RuntimeArgumentsCustomExample.java) is one such example.
+The generic defined with the `RuntimeConfigValue` interface defines what value that runtime argument should return.
+
+Once you have created your runtime argument class, use `RuntimeConfigManager.getInstance().addRuntimeArgument` to add it to the manager. Then, you can reference the value that is set during runtime by referencing the key, as so:
+
+`RuntimeConfigManager.getInstance().getRuntimeArgumentValue(CustomRuntimeArgument.KEY);`
+
+See the [associated test class](src/test/java/com/ebay/nst/tutorials/rest/runtimearguments/RuntimeArgumentsTest.java) for an example of adding a custom runtime argument.
+
+## Overriding runtime arguments
+
+As mentioned above, we have access to directly override the Platform runtime argument using `RuntimeConfigManager`'s `overridePlatform` method. 
+
+However, in order to override any other runtime arguments, we must first retrieve the `RuntimeConfigValue` that has been set.
+Because the runtime argument values can be of any data type, we will need to retrieve and cast it using `RuntimeConfigManager.getRuntimeArgument`.
```

Comment:
You may, as an alternative, retrieve the specific argument type you want to override. In this way you are holding a typed argument and the generic type specified will now be used for the getter and setter, instead of having to do any casting.

Reply:
You are doing a form of this in the snippit below.

Reply:
Done in 3d43f0926bdcdf8d1cddbe7d5466662922cf51bb